test/services/notificationHubsManagement/registration-tests.js

```diff
@@ -0,0 +1,212 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+var should = require('should');
+var azure = require('azure');
+var SuiteBase = require('../../framework/suite-base');
+var NotificationHubsManagementClient = require('../../../lib/services/notificationHubsManagement/lib/notificationHubsManagementClient');
+var testPrefix = 'notificationhubsservice-NS-tests';
+var groupPrefix = 'nodeTestGroup';
+var namespacePrefix = 'testNS';
+
+var requiredEnvironment = [
+  { name: 'AZURE_TEST_LOCATION', defaultValue: 'South Central US' }
+];
+
+var notificationHubPrefix = 'testHub';
+var notificationHubName;
+var testPrefix = 'notificationhubsservice-NS-tests';
+var groupPrefix = 'nodeTestGroup';
+var namespacePrefix = 'testNS';
+var authPrefix = 'testAuth';
+var groupName;
+var createdGroups = [];
+var createdAccounts = [];
+
+describe('Notification Hubs Management :',
+  function () {
+    before(function (done) {
+      suite = new SuiteBase(this, testPrefix, requiredEnvironment);
+      suite.setupSuite(function () {
+        groupName = suite.generateId(groupPrefix, createdGroups, suite.isMocked);
+        client = new NotificationHubsManagementClient(suite.credentials, suite.subscriptionId);
+        namespaceName = suite.generateId(namespacePrefix, createdAccounts, suite.isMocked);
+        notificationHubName = suite.generateId(notificationHubPrefix, createdAccounts, suite.isMocked);
+        authorizationRuleName = suite.generateId(authPrefix, createdAccounts, suite.isMocked);
+        namespaceLocation = process.env['AZURE_TEST_LOCATION'];
+        createNamespaceParameters = {
+          location: namespaceLocation,
+          tags: {
+            tag1: 'value1',
+            tag2: 'value2'
+          },
+          sku: { name: "Standard" }
+        };
+
+        authRuleParameter = {
+          location: namespaceLocation,
+          name: authorizationRuleName,
+          properties: {
+            rights: ['Listen', 'Send']
+          }
+        };
+
+        createNotificationHubParameters = {
+          location: namespaceLocation,
+          wnsCredential: {
+            packageSid:
+            'ms-app://s-1-15-2-1817505189-427745171-3213743798-2985869298-800724128-1004923984-4143860699',
+            secretKey: 'w7TBprR-THIS-IS-DUMMY-KEYAzSYFhp',
+            windowsLiveEndpoint: 'http://pushtestservice.cloudapp.net/LiveID/accesstoken.srf'
+          },
+          apnsCredential: {
+            token:
+            "MIGTAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBHkwdwIBAQQgpVB15km4qskA5Ra5XvdtOwWPvaXIhVVQZdonzINh + hGgCgYIKoZIzj0DAQehRANCAASS3ek04J20BqA6WWDlD6 + xd3dJEifhW87wI0nnkfUB8LDb424TiWlzGIgnxV79hb3QHCAUNsPdBfLLF + Od8yqL",
```

Comment:
is this a real token (this is a public repo)?

Reply:
This is common set of Apple Notification Service credential we use across our team for testing. This is created for our team. Also, notificationhub-tests.js uses this token